src/vcr-sharp/Cassette.cs

```diff
@@ -0,0 +1,169 @@
+﻿namespace VcrSharp
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using YamlDotNet.Serialization;
+    using YamlDotNet.Serialization.NamingConventions;
+
+    public class Cassette
+    {
+        private int currentIndex = 0;
+        private readonly string cassettePath;
+        private List<CachedRequestResponse> cachedEntries;
+        private List<CachedRequestResponse> storedEntries;
+
+        public Cassette(string cassettePath)
+        {
+            this.cassettePath = cassettePath;
+        }
+
+        async Task SetupCache()
+        {
+            if (File.Exists(cassettePath))
+            {
+                var file = new FileInfo(cassettePath);
+                using var stream = file.OpenRead();
+                using var reader = new StreamReader(stream);
+                var serializer = new DeserializerBuilder()
+                    .WithNamingConvention(CamelCaseNamingConvention.Instance)
+                    .Build();
+
+                var task = Task.Factory.StartNew(() => serializer.Deserialize<CachedRequestResponseArray>(reader));
+                // var task = Task.Factory.StartNew(() => JsonSerializer.Deserialize<CachedRequestResponseArray>(File.ReadAllText(cassettePath)));
+                var contents = await task;
+                cachedEntries = new List<CachedRequestResponse>(contents.HttpInteractions ?? Array.Empty<CachedRequestResponse>());
+            }
+            else
+            {
+                cachedEntries = new List<CachedRequestResponse>();
+            }
+
+            storedEntries = new List<CachedRequestResponse>();
+        }
+
+        static bool MatchesRequest(CachedRequestResponse cached, HttpRequestMessage request)
+        {
+            if (!string.Equals(cached.Request.Method, request.Method.Method, StringComparison.OrdinalIgnoreCase))
+            {
+                return false;
+            }
+            if (cached.Request.Uri != request.RequestUri.ToString())
+            {
+                return false;
+            }
+            return true;
+        }
```

Comment:
I think the request content should be included here - otherwise a change to the content could break something but the tests would still pass in playback or cached vcr mode.

Reply:
fixed with commit 8b92cf5